### ISSUE TITLE
Unhelpful error message

### ISSUE THREAD

**Issue by cboursnell** [kingsfordgroup/sailfish#62]
I have a large number of error messages being generated by salmon that are in the form:
```
WARNING: Detected suspicious pair --- 
	The proper-pair statuses are inconsistent:
read1 : proper-pair; mapped; matemapped
read2: no proper-pair; not mapped; matemapped
```
It would be useful if the message specified the name of the read pair that was causing the problem so that the input bam file could be checked out.

**Comment by rob-p** [comment]
Hi @cboursnell,

  Sure, I can have salmon also output the read name as well.  As the error message suggests, this occurs when the BAM flags have funny (and uninterpretable) settings.  Might you also be able to share a small example (BAM & reference) so that I can examine the "offending" reads myself?

Thanks, 
Rob

**Comment by rob-p** [comment]
Hi @cboursnell,

  I've added the names of the reads to the output in a [recent commit](https://github.com/kingsfordgroup/sailfish/commit/004c5eeaafd28c3e750cc7d2e7c1da821d47848e).  However, I'm going to postpone closing this issue or bumping the version to 0.2.4 until you've had a look to see how these reads appear in the original BAM file to ensure that the complaints made by Salmon make sense.

**Comment by cboursnell** [comment]
Thanks for adding this. I did some investigation into the bam fie. This was an issue caused by a bam file that had been incorrectly read sorted (as in it shouldn't have been sorted at all). When I used the bam straight out of the aligner I didn't get any messages from salmon. I think this is still a useful fix for anyone else in the future though. Yeah don't bump the version on account of this small change.
